nats/src/main.rs

```diff
@@ -127,38 +178,77 @@ impl NatsMessagingProvider {
             }
         };
         let url = cfg.cluster_uris.get(0).unwrap();
-        let conn = opts
+
+        let client = opts
             .connect(url)
             .await
             .map_err(|e| RpcError::ProviderInit(format!("Nats connection to {}: {}", url, e)))?;
 
+        // Connections
         for sub in cfg.subscriptions.iter().filter(|s| !s.is_empty()) {
             let (sub, queue) = match sub.split_once('|') {
                 Some((sub, queue)) => (sub, Some(queue.to_string())),
                 None => (sub.as_str(), None),
             };
-            self.subscribe(&conn, ld, sub.to_string(), queue).await?;
+
+            self.subscribe(&client, ld, sub.to_string(), queue).await?;
         }
-        Ok(conn)
+
+        Ok(client)
     }
 
     /// Add a regular or queue subscription
     async fn subscribe(
         &self,
-        conn: &async_nats::Client,
+        client: &async_nats::Client,
         ld: &LinkDefinition,
         sub: String,
         queue: Option<String>,
     ) -> RpcResult<()> {
-        let mut subscription = match queue {
-            Some(queue) => conn.queue_subscribe(sub.clone(), queue).await,
-            None => conn.subscribe(sub.clone()).await,
+        let mut subscriber = match queue {
+            Some(queue) => client.queue_subscribe(sub.clone(), queue).await,
+            None => client.subscribe(sub.clone()).await,
         }
         .map_err(|e| {
             error!(subject = %sub, error = %e, "error subscribing subscribing");
             RpcError::Nats(format!("subscription to {}: {}", sub, e))
         })?;
+
+        // Create channel for sending control messages to the NATS listen loop for a subscriber
+        let (ctrl_tx, mut ctrl_rx) = mpsc::channel(NATS_LISTEN_LOOP_CTRL_MESSAGE_BUFFER_SIZE);
+
+        // Save a copy of the sub in the map so it can be unsubscribed when the client is dropped
+        let mut _actor_map = self.actors.write().await;
```

Comment:
Nit: Don't preface with `_` unless you aren't going to use it

Reply:
woops yeah will fix this, totally right

Reply:
Removed `_`